### PR TITLE
Charmis/swift conversion

### PR DESCRIPTION
**Pull request by CharmisG**
plugin compileSDK and targetSDk Upgradation to 30

### CODE REVIEW

File: android/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -16,5 +16,19 @@ limitations under the License.
...
           package="com.squareup.sdk.reader.react">
+          
+    <uses-permission
+        android:name="android.permission.BLUETOOTH"
+        android:maxSdkVersion="30" />
+    <uses-permission
+        android:name="android.permission.BLUETOOTH_ADMIN"
+        android:maxSdkVersion="30" />
+    <!-- API 31+ -->
+    <uses-permission android:name="android.permission.BLUETOOTH_CONNECT" />
```
[the remark this answers is not in the document]
Comment (CharmisG) [reply android/src/main/AndroidManifest.xml:27]: Yes.

File: reader-sdk-react-native-quickstart/android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -24,6 +24,13 @@ limitations under the License.
     <uses-permission android:name="android.permission.BLUETOOTH_SCAN" />
     <uses-permission android:name="android.permission.CAMERA" />
 
+    <uses-permission
```
[the remark this answers is not in the document]
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/android/app/src/main/AndroidManifest.xml:27]: Yes if i remove it from quickstart then also it's crashing in about android 12 so.

File: reader-sdk-react-native-quickstart/android/build.gradle [hunk 3]
```diff
@@ -25,7 +25,7 @@ buildscript {
     }
     ext {
         buildToolsVersion = "30.0.3"
-        minSdkVersion = 23
+        minSdkVersion = 24
         compileSdkVersion = 32
         targetSdkVersion = 32
```
[the remark this answers is not in the document]
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/android/build.gradle:30]: Done

File: reader-sdk-react-native-quickstart/android/gradle.properties [hunk 4]
```diff
@@ -38,14 +38,16 @@ android.useAndroidX=true
...
 
-android.enableJetifier=true
+android.enableJetifier=false
 android.jetifier.ignorelist=moshi-1.13.0
-android.jetifier.blacklist=moshi-1.13.0
+#android.jetifier.blacklist=moshi-1.13.0
+android.jetifier.blacklist = bcprov-jdk15on
 # Version of flipper SDK to use with React Native
 FLIPPER_VERSION=0.99.0
+org.gradle.jvmargs=-Xmx1536M --add-exports=java.base/sun.nio.ch=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED
```
[the remark this answers is not in the document]
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/android/gradle.properties:47]: Done

File: reader-sdk-react-native-quickstart/android/gradle.properties [hunk 5]
```diff
@@ -38,14 +38,16 @@ android.useAndroidX=true
 org.gradle.jvmargs=-Xmx4g
 # Automatically convert third-party libraries to use AndroidX
 
-android.enableJetifier=true
+android.enableJetifier=false
 android.jetifier.ignorelist=moshi-1.13.0
-android.jetifier.blacklist=moshi-1.13.0
+#android.jetifier.blacklist=moshi-1.13.0
```
[the remark this answers is not in the document]
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/android/gradle.properties:43]: Done

File: reader-sdk-react-native-quickstart/app/screens/CheckoutScreen.tsx [hunk 6]
```diff
@@ -20,8 +20,10 @@ import {
   Text,
   Alert,
   Platform,
+  Image,
   ActionSheetIOS, // eslint-disable-line react-native/split-platform-components
 } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler';
```
[the remark this answers is not in the document]
Comment (CharmisG) [reply reader-sdk-react-native-quickstart/app/screens/CheckoutScreen.tsx:26]: Sure please ignore this i will add this in the new PR.

**Review by hukid — APPROVED** [review]

**Comment by hukid** [comment]
Don't you need permission changes in this PR?

- [ ] double check if permission changes are missing?

**Pull request merged**